common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -1407,8 +1413,7 @@ public AccountDeletionRecord removeAccount(@Nullable final String environment,
         );
 
         final AccountRecord targetAccount;
-        if (null == environment
-                || null == clientId
+        if (null == clientId
```

Comment:
Doesn't make sense to me why the check for environment is removed? I think the check needs to stay there and instead the caller should supply `null` I believe to delete records from all environments, right?

Reply:
With this check if environment is passed as null, remove account is no-op and returns early, without removing anything.

Reply:
I think we should make the change **_there_** then

Reply:
@iambmelt Feel free to add your thoughts as well as you are the cache expert

Reply:
This the place to remove the check, remove account happens after this check in this method

Reply:
Ok that makes sense.....hadn't looked at the rest of the code earlier......so yeah this check had to be removed then. Thanks for clarifying.